recipes/hyperscan/all/conanfile.py

```diff
@@ -0,0 +1,89 @@
+from conans import ConanFile, CMake, tools
+
+import os
+
+class HyperscanConan(ConanFile):
+    name = "hyperscan"
+    license = "BSD-2-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.hyperscan.io"
+    description = "High-performance regular expression matching library"
+    topics = ("regex", "regular expressions")
+    settings = "os", "compiler", "build_type", "arch"
+    export_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+
+    _cmake = None
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "optimise": [True, False],
+        "debug_output": [True, False],
+        "build_avx512": [True, False],
+        "fat_runtime": [True, False],
+        "build_chimera": [True, False],
+        "dump_support": [True, False]
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "optimise": True,
+        "debug_output": False,
+        "build_avx512": False,
+        "fat_runtime": False,
+        "build_chimera": False,
+        "dump_support": False
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("hyperscan-{0}".format(self.version), self._source_subfolder)
+
+    def build_requirements(self):
+        self.build_requires("boost/1.73.0");
+        self.build_requires("ragel/6.10");
+
+    def requirements(self):
+        if self.options.build_chimera:
+            self.required("pcre/8.41")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self);
+        self._cmake.definitions["OPTIMISE"] = self.options.optimise
+        self._cmake.definitions["DEBUG_OUTPUT"] = self.options.debug_output
+        self._cmake.definitions["BUILD_AVX512"] = self.options.build_avx512
+        self._cmake.definitions["FAT_RUNTIME"] = self.options.fat_runtime
+        self._cmake.definitions["BUILD_CHIMERA"] = self.options.build_chimera
+        self._cmake.definitions["DUMP_SUPPORT"] = self.options.dump_support
+        self._cmake.configure(source_folder=self._source_subfolder, build_folder=self._build_subfolder)
```

Comment:
```suggestion
        self._cmake.configure(build_folder=self._build_subfolder)
```

Reply:
Removing the `source_folder` parameter makes the install step fail, since CMake looks for the generated .pc file in the wrong place. I'm still investigating.

Reply:
See https://github.com/intel/hyperscan/blob/c00683d73916e39f01b0d418f686c8b5c379159c/CMakeLists.txt#L500-L501

`CMAKE_BINARY_DIR` must be `CMAKE_CURRENT_BINARY_DIR`.

I think you a patch is required to replace all `CMAKE_BINARY_DIR` with `CMAKE_CURRENT_BINARY_DIR`, and `CMAKE_SOURCE_DIR` with `CMAKE_CURRENT_SOURCE_DIR`.

Also, [settings CMAKE_MODULE_PATH](https://github.com/intel/hyperscan/blob/c00683d73916e39f01b0d418f686c8b5c379159c/CMakeLists.txt#L9) is a big no no.
It should be added with `list(APPEND)` and the uses of `CMAKE_MODULE_PATH` should be replaced with some other variable.